Add docs for packaging changes

dedemorton opened the pull request:
Documents new directory structure, settings files, and instructions for running Logstash as a service.

This content will appear in the following location.

![settinguporg](https://cloud.githubusercontent.com/assets/14206422/15580789/c1a23cea-231e-11e6-8ee1-90eaa42a3bf5.png)

untergeek commented on docs/static/setting-up-logstash.asciidoc:
```diff
@@ -0,0 +1,183 @@
+| conf
+  | Configuration files, including `logstash.yml` and `jvm.options`
+  | `{extract.path}/config`
+  | `path.conf`
```
This should be `path.settings`

untergeek commented on docs/static/setting-up-logstash.asciidoc:
```diff
@@ -0,0 +1,183 @@
+  | `{extract.path}/bin`
+ d|
+
+| conf
```
This should be "settings"

untergeek commented on docs/static/setting-up-logstash.asciidoc:
```diff
@@ -0,0 +1,183 @@
+  | `/usr/share/logstash/bin`
+ d|
+
+| conf
```
This should be "settings"

untergeek commented on docs/static/setting-up-logstash.asciidoc:
```diff
@@ -0,0 +1,183 @@
+| conf
+  | Configuration files, including `logstash.yml`, `jvm.options`, and `startup.options`
+  | `/etc/logstash`
+  | `path.conf`
```
This should be `path.settings`

untergeek commented on docs/static/setting-up-logstash.asciidoc:
```diff
@@ -0,0 +1,183 @@
+| conf
+  | Logstash pipeline configuration files
+  | `/etc/logstash/conf.d`
+ d|
```
This should be `path.conf`

untergeek commented on docs/static/setting-up-logstash.asciidoc:
```diff
@@ -0,0 +1,183 @@
+  service name, and service description. By default, Logstash services are installed under the user `logstash`. The `startup.options` file makes it easier for you to install multiple instances of the Logstash service. You can copy
+  the file and change the values for specific settings. Note that the `startup.options` file is not read at startup. If
+  you want to change the Logstash startup script (for example, to change the Logstash user or read from a different
+  configuration path), you must re-run the `system-install` script (as root) and pass in the new settings.
```
"and pass" should read "to pass"

untergeek commented:
Other than the inline comments, this looks good!

untergeek commented on docs/static/setting-up-logstash.asciidoc:
```diff
@@ -0,0 +1,183 @@
+| conf
+  | Logstash pipeline configuration files
+  | `/etc/logstash/conf.d`
+  | `path.conf`
```
My bad.  This should be `path.config` as the defaults in `logstash.yml` make plain:
```
# ------------ Pipeline Configuration Settings --------------
#
# Where to fetch the pipeline configuration for the main pipeline
#
# path.config:
```

dedemorton commented:
@untergeek @suyograo I think I've addressed all concerns. Can I squash the commits and merge?

One last thing, tho: regarding the TOC, I think it makes sense to move the topics about the CLI and settings file to the new section. Here's what it would look like:

![settinguptoc](https://cloud.githubusercontent.com/assets/14206422/15584363/2eb00d2a-232f-11e6-96ac-8cc15e9a4f99.png)

Let me know what you think.

suyograo commented:
@dedemorton LGTM from my side..

suyograo commented:
+1 on moving to a new section
